pkg/kube/controllers/extendedjob/runner.go

```diff
@@ -72,19 +88,37 @@ func (r *RunnerImpl) Run() {
 
 	for _, extJob := range extJobs.Items {
 		filtered := r.query.Match(extJob, podEvents)
+		// if multiple events belong to the same pod `query.Match` will return
+		// all of the (event, pod) tuples.
+		// However we update the timestamp in the first iteration of
+		// this loop. Further iterations won't run the same job twice
+		// on the same pod, since the updated timestamp will be newer
+		// than any event we look at in this loop.
 		for _, podEvent := range filtered {
-			if r.isFreshStamp(extJob, podEvent) {
+			// this might be too noisy
+			r.log.Debugf("%s: looking at pod event: %s/%s for pod %s", extJob.Name, podEvent.Event.Name, podEvent.Event.Reason, podEvent.podName())
+
+			if podEvent.shouldTrigger(extJob.Name) {
```

Comment:
I'm confused why this is not `if !podEvent.shouldTrigger(extJob.Name)`. Is this actually correct?

Reply:
yes